lib/_http_client.js

```diff
@@ -140,13 +151,31 @@ function ClientRequest(options, cb) {
     self._last = true;
     self.shouldKeepAlive = false;
     if (options.createConnection) {
-      self.onSocket(options.createConnection(options));
+      options.oncreate = oncreate;
```

Comment:
No, the user's `options` is merged/extended with a new object at the beginning of this constructor.